Common/src/containers/CTrapezoidalMap.cpp

```diff
@@ -174,7 +175,9 @@ pair<unsigned long, unsigned long> CTrapezoidalMap::GetBand(su2double val_x) {
 
   std::pair<std::vector<su2double>::iterator,std::vector<su2double>::iterator> bounds;
   bounds = std::equal_range (unique_bands_x.begin(), unique_bands_x.end(), val_x);
-  i_up =  bounds.first - unique_bands_x.begin();
+
+  /*--- if upper bound = 0, then use the range [0,1] ---*/
+  i_up =  max((unsigned long)(1), (unsigned long)(bounds.first - unique_bands_x.begin()));
```

Comment:
Don't use c-style casts please.
```suggestion
  i_up =  max<unsigned long>(1, bounds.first - unique_bands_x.begin());
```